doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+
+# Functional requirements
+
+### As Signal Consumer
+1. As a client I want to have a single service point accessible from my execution environment.
+   
+2. As a client I want to `Get` sensor values or actuator current values of the vehicle and its environment in a uniform manner.
+  - Details:
+    * All sensor and actuator values for which a client have permission to.
+    * Client must provide the paths of the requested signals, which optionally may contain wildcards.
+    * I want to receive exactly all the signals’ values I requested without errors.
+  
+3. As a client I want to `Subscribe` to sensor values or actuator current values when some signal value changed -> what about continuous subscription?
+  - Details:
+    * All sensor and actuator values for which a client have permission to.
+    * Client must provide the path, paths or wildcard paths.
+    * I want to receive exactly all the signals’ values I requested without errors.
+
+4. As a client I want to `Subscribe` and be able to apply a filter to receive a sensor value with an interval of x milliseconds only if the value has changed.
+   
+5.  As a client I want to `GetMetadata` static Metadata from signals.
+  - Details:
+    * All sensors, actuators, and attributes values for which a client have permission to.
+    * Client must provide the path, paths or wildcard.
+
+6.  As a client I want to `Get` attributes values once.
+
+7.  As a client I want to `Actuate` a target value of an actuator and guarantee that the operation succeeded.
+
+8.  As a client I want to connect and interact with Databroker either if it contains data or not.
+
+10.  As a client I want to know if there is a provider up and running by getting some sensor or actuator value.
+
+11.  As a client I don’t want to `Subscribe` to target values.
+
+12.  As a client I don’t want to `Subscribe` to attributes values.
+
+13.  As a client I don’t want to `Get` target values.
+
+14.  As a client I don’t want to `Set` sensor values.
+
+15.  As a client I don’t want to `Set` attributes values.
+
+### As Databroker
+1. The data broker shall be able to handle catalogs of signals described by the syntax as defined by the COVESA Vehicle Signal Specification (VSS). This relates to all aspects of the VSS syntax definition, which is also called VSS rule set. This implies that the data broker can handle the signal catalog as defined by the COVESA VSS.
+
+2. Databroker shall support at least those metadata elements as defined by the VSS rule set.
+
+### As Provider
+1. As a provider I want to `Set` the value and current value of one or many sensors and/or actuators with its timestamp to Databroker.
+2. 
+3. As a provider I want to `Update` using stream atomically the value and current value of one or many sensors and/or actuators with its timestamp to Databroker.
+
+4. As a provider I want to `Subscribe` to actuator target values requests.
+
+5. As a provider I want to `Set` once attributes values.
```

Comment:
Yes, attribute values could be part of the signal config (vss.json) passed to the broker at startup.

Question is then: How to update databroker with new signals at runtime (e.g. in case of a "dynamic" update)? Easiest option would be to stop and restart the broker with the new signal config.

Reply:
I agree that regular providers do not handle attribute values. If there is a requirement to change attributes during runtime, this might be an argument for this attribute actually being a sensor. 

We had a number of concerns with changing the VSS tree during runtime (adding/removing nodes etc.), so I would changes to attributes only through the signal config at the start (as proposed by @BjoernAtBosch )